src/app/cli/interpreter.cpp

```diff
@@ -1432,20 +1433,32 @@ Interpreter::Value Interpreter::ProcessBr(const Expression &aExpr)
         nlohmann::json                            baJson;
         char                                      mdnsSendBuffer[kMdnsBufferSize];
 
-        if (mContext.mCommandKeys.size() == 2 && mContext.mCommandKeys[0] == "--timeout")
-        {
+        auto it = std::find(mContext.mCommandKeys.begin(), mContext.mCommandKeys.end(), "--timeout");
+
+        if (it != mContext.mCommandKeys.end()) {
```

Comment:
This will crash if the user provides only "--timeout", e.g. "br scan --timeout".
```suggestion
        if (it != mContext.mCommandKeys.end() && it +1 != mContext.mCommandKeys.end()) {
```

Reply:
The same for the "--netif" option